dali/operators/decoder/video/video_decoder_mixed.cc

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/operators/decoder/video/video_decoder_mixed.h"
+
+namespace dali {
+
+template <>
+bool VideoDecoderBase<MixedBackend, FramesDecoderGpu>::SetupImpl(
+  std::vector<OutputDesc> &output_desc, const workspace_t<MixedBackend> &ws) {
+  ValidateInput(ws);
+  const auto &input = ws.Input<CPUBackend>(0);
+  int batch_size = input.num_samples();
+  auto stream = ws.stream();
+  frames_decoders_.resize(batch_size);
+  for (int i = 0; i < batch_size; ++i) {
+    auto sample = input[i];
+    auto data = reinterpret_cast<const char *>(sample.data<uint8_t>());
+    size_t size = sample.shape().num_elements();
```

Comment:
I would add an enforce that it can't be == 0.

Reply:
Added this check to ValidateInput. This covers CPU variant as well